xrpl/transaction/main.py

```diff
@@ -74,20 +83,19 @@ def safe_sign_transaction(
     Returns:
         The signed transaction blob.
     """
-    if check_fee:
-        _check_fee(transaction)
-    transaction_json = _prepare_transaction(transaction, wallet)
-    serialized_for_signing = encode_for_signing(transaction_json)
-    serialized_bytes = bytes.fromhex(serialized_for_signing)
-    signature = sign(serialized_bytes, wallet.private_key)
-    transaction_json["TxnSignature"] = signature
-    return cast(Transaction, Transaction.from_xrpl(transaction_json))
+    return asyncio.run(
+        main.safe_sign_transaction(
+            transaction,
+            wallet,
+            check_fee,
+        )
+    )
```

Comment:
Does this even need an `asyncio.run` around it? It should still be a sync function originally

Reply:
it does, because of `check_fee`